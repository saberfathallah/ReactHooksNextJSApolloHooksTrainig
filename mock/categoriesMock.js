export const CATEGORY = {
    id: "5e89c5f78245cd0d75dec50a",
    level: 1,
    name: "Sport",
    children: [
      {
        children: [],
        id: "5e89c6068245cd0d75dec50c",
        name: "foot",
        level: 2,
      },
    ],
  }