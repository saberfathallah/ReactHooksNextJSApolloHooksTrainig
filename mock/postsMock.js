export const POST = {
  categoryId: "5e89c6458245cd0d75dec50f",
  postId: "5e89c6458245cd0d75dec50f",
  comments: [],
  creatorId: "5e89c2fd8245cd0d75dec501",
  description: "corona france post",
  likes: ["5e8a26784c6b1d234b4c1a3f", "5e89c2fd8245cd0d75dec501"],
  id: "5e89c6878245cd0d75dec510",
  userId: {
    id: "5fac3671dd9b1b079249d522",
    email: "sberrr@gmail.com",
    name: "sberrr",
  },
};

// interface Post {
//   userId: User;
//   categoryId: string;
//   postId: string;
//   comments: Comment[];
//   description: string;
//   id: string;
//   likes: string[];
// }

// categoryId: "5e89c6458245cd0d75dec50f"
// comments: (5) [{…}, {…}, {…}, {…}, {…}]
// description: "corona france post"
// id: "5e89c6878245cd0d75dec510"
// likes: (2) ["5e8a26784c6b1d234b4c1a3f", "5e89c2fd8245cd0d75dec501"]
// userId: {name: "Saber Fathallah", id: "5e89c2fd8245cd0d75dec501", __typename: "User"}
