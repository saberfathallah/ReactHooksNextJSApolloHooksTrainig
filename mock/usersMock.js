export const USER_CONNECTED = {
    id: "5fac3671dd9b1b079249d522",
    email: "sberrr@gmail.com",
    name: "sberrr",
  };