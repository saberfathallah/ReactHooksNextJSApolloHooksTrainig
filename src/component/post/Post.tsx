import React from 'react';
import { createStyles, Theme, makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MoreVertIcon from '@material-ui/icons/MoreVert';

import Comments from '../comments';
import Like from '../like';

const useStyles = makeStyles((theme: Theme) => createStyles({
  card: {
    maxWidth: 500,
    margin: theme.spacing(2),
  },
  media: {
    height: 250,
  },
  root: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  heart: {
    float: 'right',
  },
  textField: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
    width: '25ch',
  },
}));

interface User {
  name: string;
  id: string;
}

interface Comment {
  userId: User;
  description: string;
  postId: string;
  id: string;
}
interface PostProps {
  categoryId: string;
  postId: string;
  userName: string;
  description: string;
  comments: Comment[];
  userConnected: User;
  creatorId: string;
  likes: string[];
}

const Post: React.FC<PostProps> = (props: PostProps) => {
  const {
    description, comments, userName, categoryId, postId, creatorId, userConnected, likes,
  } = props;
  const classes = useStyles();

  return (
    <Card className={classes.card}>
      <CardHeader
        avatar={(
          <Avatar
            alt="Ted talk"
            src="https://pbs.twimg.com/profile_images/877631054525472768/Xp5FAPD5_reasonably_small.jpg"
          />
        )}
        action={
          userConnected.id === creatorId && (
            <IconButton aria-label="settings">
              <MoreVertIcon />
            </IconButton>
          )
        }
        title={userName}
        subheader="5 hours ago"
      />
      <CardContent>
        <Typography variant="body2" color="textSecondary" component="p">
          {
            description
          }
        </Typography>
        <Like likes={likes} userConnected={userConnected} />
        <Comments
          creatorId={creatorId}
          userConnected={userConnected}
          comments={comments}
          categoryId={categoryId}
          postId={postId}
        />
      </CardContent>
    </Card>
  );
};

export default Post;
