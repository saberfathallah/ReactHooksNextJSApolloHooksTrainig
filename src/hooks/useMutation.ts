import { useMutation as useMutationApollo } from "@apollo/react-hooks";

export const useMutation = (
  mutation: any,
  { variables, optionCache, ...otherOptions }: any
): any => {
  if (optionCache) {
    const [mutationFunction, { data, error, loading }] = useMutationApollo(
      mutation,
      {
        variables,
        update: (cache, { data }) =>
          optionCache.updateCacheFunction(cache, data, optionCache.variables),
          ...otherOptions,
      }
    );
    return [mutationFunction, { data, error, loading }];
  }

  const [mutationFunction, { data, error, loading }] = useMutationApollo(
    mutation,
    {
      variables,
    }
  );

  return [mutationFunction, { data, error, loading }];
};
