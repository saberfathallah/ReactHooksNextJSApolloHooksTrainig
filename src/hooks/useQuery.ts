import { useQuery as useQueryApollo } from "@apollo/react-hooks";

export const useQuery = (query, options={}): any => {
  const { loading, data, ...rest } = useQueryApollo(query, {
    ...options,
  });

  return { data, loading, ...rest };
}

// export default useQuery;

