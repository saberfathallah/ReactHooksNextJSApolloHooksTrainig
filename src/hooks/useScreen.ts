import { useEffect, useState } from "react";
import { isMobile, isMobileWidth } from "@utils/isBrowser";
import { ESKTOP_BREAKPOINT } from "@constants/dimensions";

const useScreen = (isClient) => {
  const [screen, setScreen] = useState({
    screenWidth: ESKTOP_BREAKPOINT,
    isMobileSize: isMobile,
  });

  useEffect(() => {
    if (!isClient) {
      return;
    }

    // first load
    setScreen({
      screenWidth: window.innerWidth,
      isMobileSize: isMobileWidth(window.innerWidth),
    });

    const handleResize = () => {
      if (isClient) {
        setScreen({
          screenWidth: window.innerWidth,
          isMobileSize: isMobileWidth(window.innerWidth),
        });
      }
    };

    window.addEventListener("resize", handleResize);

    return () => window.removeEventListener("resize", handleResize);
  }, [isClient]);

  return screen;
};

export default useScreen;
