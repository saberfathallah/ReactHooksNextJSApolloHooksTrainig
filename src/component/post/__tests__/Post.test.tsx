import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { mount } from "enzyme";

import Post from "../Post";
import { useQuery } from "../../../hooks/useQuery";
import { useMutation } from "../../../hooks/useMutation";
import { USER_CONNECTED } from "../../../../mock/usersMock";

jest.mock("../../../hooks/useQuery", () => ({
  useQuery: jest.fn(),
}));

jest.mock("../../../hooks/useMutation", () => ({
  useMutation: jest.fn(),
}));

// @ts-ignore
useQuery.mockImplementation(() => ({ data: "ddd" }));
// @ts-ignore
useMutation.mockImplementation(() => [() => {}, { loading: false }]);

const renderer = new ShallowRenderer();

describe("<Post />", () => {
  const props = {
    categoryId: "5e89c6458245cd0d75dec50f",
    comments: [],
    creatorId: "5e89c2fd8245cd0d75dec501",
    description: "corona france post",
    likes: ["5e8a26784c6b1d234b4c1a3f", "5e89c2fd8245cd0d75dec501"],
    postId: "5e89c6878245cd0d75dec510",
    userConnected: USER_CONNECTED,
    userName: "Saber Fathallah",
  };

  it("should match snapshot", () => {
    const tree = renderer.render(<Post {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("should display post", () => {
    const wrapper = mount(<Post {...props} />);
    expect(wrapper.text()).toContain("corona france post");
  });
});
