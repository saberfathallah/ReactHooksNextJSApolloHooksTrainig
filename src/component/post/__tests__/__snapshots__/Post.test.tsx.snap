// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should match snapshot 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-card-1"
>
  <WithStyles(ForwardRef(CardHeader))
    action={false}
    avatar={
      <WithStyles(ForwardRef(Avatar))
        alt="Ted talk"
        src="https://pbs.twimg.com/profile_images/877631054525472768/Xp5FAPD5_reasonably_small.jpg"
      />
    }
    subheader="5 hours ago"
    title="Saber Fathallah"
  />
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
      component="p"
      variant="body2"
    >
      corona france post
    </WithStyles(ForwardRef(Typography))>
    <LikeContainer
      likes={
        Array [
          "5e8a26784c6b1d234b4c1a3f",
          "5e89c2fd8245cd0d75dec501",
        ]
      }
      postId="5e89c6878245cd0d75dec510"
      userConnected={
        Object {
          "email": "sberrr@gmail.com",
          "id": "5fac3671dd9b1b079249d522",
          "name": "sberrr",
        }
      }
    />
    <CommentsContainer
      categoryId="5e89c6458245cd0d75dec50f"
      comments={Array []}
      creatorId="5e89c2fd8245cd0d75dec501"
      postId="5e89c6878245cd0d75dec510"
      userConnected={
        Object {
          "email": "sberrr@gmail.com",
          "id": "5fac3671dd9b1b079249d522",
          "name": "sberrr",
        }
      }
    />
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
